reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscriber.scala

```diff
@@ -53,7 +55,14 @@ final class StreamSubscriber[F[_]: ConcurrentEffect, A](val sub: StreamSubscribe
 
 object StreamSubscriber {
   def apply[F[_]: ConcurrentEffect, A]: F[StreamSubscriber[F, A]] =
```

Comment:
I'm unsure of the overloading here, but I'd prefer not to break API.

Reply:
```scala
  def apply[F[_]: ConcurrentEffect, A](batchSize: Option[Int] = None): F[StreamSubscriber[F, A]] =
```
I _think_ this'll give us backwards compatibility, but I'd have to confirm.

Thought, I don't like the default argument here, as the default argument is an artifact of the previous iteration of the code rather than a design decision.  So another idea is that we keep `def apply` as is, and then introduce 
```scala
def applyBatch[F[_]: ConcurrentEffect, A](batchSize: Int): F[StreamSubscriber[F, A]]
```